generators/newmodel/index.js

```diff
@@ -0,0 +1,163 @@
+const Generator = require('yeoman-generator');
+const fs = require('fs');
+
+module.exports = class extends Generator {
+
+	prompting() {
+		if (this.options.isSubgeneratorCall) {
+			this.destinationRoot(this.options.cwd);
+			this.options.oneTimeConfig = this.config.getAll();
+			return;
+		}
+		var aPrompt = [{
+				type: 'input',
+				name: 'modelName',
+				message: 'What is the name of your model, press enter if it is the default model?',
+				validate: (s) => {
+					if (/^[a-zA-Z0-9_-]*$/g.test(s)) {
+						return true;
+					}
+					return 'Please use alpha numeric characters only for the view name.';
+				},
+				default: ""
+			},
+			{
+				type: 'list',
+				name: 'modelType',
+				message: 'Which type of model do you want to add?',
+				choices: ['OData', 'JSON'],
+				default: 'OData'
+			},
+			{
+				type: 'list',
+				name: 'bindingMode',
+				message: 'Which binding mode do you want to use?',
+				choices: ['TwoWay', 'OneWay'],
+				default: 'TwoWay'
+			},
+			{
+				when: function (props) {
+					return props.modelType === "OData";
+				},
+				type: 'input',
+				name: 'url',
+				message: 'What is the data source url'
+			},
+			{
+				when: function (props) {
+					return props.modelType === "OData";
+				},
+				type: 'list',
+				name: 'countMode',
+				message: 'Which count mode do you want to use?',
+				choices: ['Inline', 'Request'],
+				default: 'Inline'
+			}
+		];
+
+
+
+		return this.prompt(aPrompt).then((answers) => {
+			this.options.oneTimeConfig = this.config.getAll();
+			this.options.oneTimeConfig.modelName = answers.modelName;
+			this.options.oneTimeConfig.modelType = answers.modelType;
+			this.options.oneTimeConfig.bindingMode = answers.bindingMode;
+			if (answers.modelType === 'OData') {
+
+				this.options.oneTimeConfig.url = answers.url;
+				this.options.oneTimeConfig.countMode = answers.countMode;
+				console.log(this.options.oneTimeConfig.countMode)
```

Comment:
The [documentation suggests](https://yeoman.io/authoring/user-interactions.html) that we shouldn't use the console object in yeoman